python/ray/train/tests/test_predictor.py

```diff
@@ -45,27 +53,17 @@ def test_from_checkpoint():
     assert DummyPredictor.from_checkpoint(checkpoint).factor == 2.0
 
 
-@mock.patch(
-    "ray.train.predictor.convert_batch_type_to_pandas",
-    return_value=mock.DEFAULT,
-)
-@mock.patch(
-    "ray.train.predictor.convert_pandas_to_batch_type",
-    return_value=mock.DEFAULT,
-)
-def test_predict(convert_from_pandas_mock, convert_to_pandas_mock):
-    checkpoint = Checkpoint.from_dict({"factor": 2.0})
+def test_predict():
+    checkpoint = Checkpoint.from_dict(
+        {"factor": 2.0, PREPROCESSOR_KEY: DummyPreprocessor()}
+    )
     predictor = DummyPredictor.from_checkpoint(checkpoint)
 
     input = pd.DataFrame({"x": [1, 2, 3]})
     expected_output = input * 4.0
     actual_output = predictor.predict(input)
     assert actual_output.equals(expected_output)
 
-    # Ensure the proper conversion functions are called.
-    convert_to_pandas_mock.assert_called_once()
-    convert_from_pandas_mock.assert_called_once()
```

Comment:
Removing these mocks since they cause the prediction logic to no longer be tested. Didn't seem valuable enough of a test to create a second test that just tests the call logic.

Reply:
Hmm I wrote the mocks after our previous discussions on not writing integration tests for everything.

If we make sure the conversion functions are called and since the conversion functions are also tested, then we don’t need to write tests for predictor against all possible input and output types

Reply:
Ah gotcha. I think in that case I can extend the mocks to return concrete values and validate the inputs/outputs.